### ISSUE TITLE
[Question] Running training on all data

### ISSUE THREAD

**Issue by machur** [MIC-DKFZ/nnDetection#143]
Hi @mibaumgartner, I would like to run a training on all the training data (without cross-validation and splitting into folds). I created my own splits_final.pkl file with only one fold defined (containing whole training set and an empty validation set), but I got some errors - it looks like the validation set is required anyway. What would be the easiest way to run such setup? Do you have any recommendations?

**Comment by machur** [comment]
I was wondering about hacking the validation set somehow, but I want to make sure that it will not be used for sweeping/model fine-tuning in any way.

**Comment by mibaumgartner** [comment]
Dear @machur, there is no intended way of doing this, so the only option is to dig into the code and manually change around the splits and code.

Some points that come to mind:
- Update the splits file to only contain the training
- Manually change the train script to return None for the validation datamodule during training => that should skip validation during training
- Remove sweep and eval after calling the training function

**Comment by machur** [comment]
That's what I suspected. Thanks for the quick response!

**Comment by code8865** [comment]
Dear Michael,

I was wondering if there has been an update on this matter in the meantime, or if there still requires intervention in the code. Would it perhaps be an option to simply validate on the training data, by listing the full dataset in both 'train' and 'val' in the "splits_final.pkl"?

**Comment by mibaumgartner** [comment]
Dear @code8865 ,

yes, you could simply add some cases in for the validation. Note, that the validation shown during training will be significantly overestimating the performance and there might be some caveats when determining the postprocessing parameters since the predictions will be more accurate than in reality. Still worth a try :) 

Best,
Michael

**Comment by tbskbr** [comment]
Hello, 

I need to deploy a unique model as well in prod so this question is of interest. 

> Dear @code8865 ,
> 
> yes, you could simply add some cases in for the validation. Note, that the validation shown during training will be significantly overestimating the performance and there might be some caveats when determining the postprocessing parameters since the predictions will be more accurate than in reality. Still worth a try :)
> 
> Best, Michael

@mibaumgartner  related to this, what do you think of 

1. Doing everything by default with 5-fold CV,
2. Realizing the consolidation,
then,
3. Saving all the sweep, consolidate, etc. parameters from steps 1 and 2, 
4. Creating fake/artificial data for validation,
5. Training a fold with whole dataset in training and fake/artificial data as validation, 
6. Ignoring sweep and everything,
7. Deploy the model as if it was the ensembled one from consolidation.

If you think this approach sounds ok, do you see some potentials difficulties or traps to realize it? Do you have hints. 

Thank you very much.
